WordPress/Classes/Models/MenuItem.h

```diff
@@ -1,101 +1,109 @@
 
 /**
  *  @brief    An object encapsulating an individual Menu item and it's API data.
  */
 @interface MenuItem : NSManagedObject
 
-@property (nullable, nonatomic, retain) NSString *contentId;
-@property (nullable, nonatomic, retain) NSString *details;
-@property (nullable, nonatomic, retain) NSString *itemId;
-@property (nullable, nonatomic, retain) NSString *linkTarget;
-@property (nullable, nonatomic, retain) NSString *linkTitle;
-@property (nullable, nonatomic, retain) NSString *name;
-@property (nullable, nonatomic, retain) NSString *type;
-@property (nullable, nonatomic, retain) NSString *typeFamily;
-@property (nullable, nonatomic, retain) NSString *typeLabel;
-@property (nullable, nonatomic, retain) NSString *urlStr;
+@property (nullable, nonatomic, strong) NSNumber *contentID;
+@property (nullable, nonatomic, strong) NSString *details;
+@property (nullable, nonatomic, strong) NSNumber *itemID;
+@property (nullable, nonatomic, strong) NSString *linkTarget;
+@property (nullable, nonatomic, strong) NSString *linkTitle;
+@property (nullable, nonatomic, strong) NSString *name;
+@property (nullable, nonatomic, strong) NSString *type;
+@property (nullable, nonatomic, strong) NSString *typeFamily;
+@property (nullable, nonatomic, strong) NSString *typeLabel;
+@property (nullable, nonatomic, strong) NSString *urlStr;
 
 ///---------------------
 /// @name Relationships
 ///---------------------
 
-/* The Menu the item belongs to.
+/**
+ The Menu the item belongs to.
  */
-@property (nullable, nonatomic, retain) Menu *menu;
+@property (nullable, nonatomic, strong) Menu *menu;
 
-/* Direct children of the item.
+/**
+ Direct children of the item.
  NOTE: Does not include any descendents of this item's children.
  See method isDescendantOfItem: for detecting ancestry.
  */
-@property (nullable, nonatomic, retain) NSSet<MenuItem *> *children;
+@property (nullable, nonatomic, strong) NSSet<MenuItem *> *children;
 
-/* The parent of the item, if the item has one.
+/**
+ The parent of the item, if the item has one.
  */
-@property (nullable, nonatomic, retain) MenuItem *parent;
+@property (nullable, nonatomic, strong) MenuItem *parent;
 
 ///---------------------
 /// @name Helper methods
 ///---------------------
-
 + (NSString *)entityName;
 
 /**
- *  @brief      Call this method to know whether an item is a descendent of this item.
- *
- *  @returns    YES if the item is a descendent, NO if not.
+ The display label text for a MenuItemType string.
+ */
++ (NSString *)labelForType:(NSString *)itemType blog:(nullable Blog *)blog;
+
+/**
+ The localized default title text (name) to use with a newly created MenuItem.
+ */
++ (NSString *)defaultItemNameLocalized;
+
+/**
+ Call this method to know whether an item is a descendent of this item.
+ @returns YES if the item is a descendent, NO if not.
  */
 - (BOOL)isDescendantOfItem:(MenuItem *)item;
 
+- (MenuItem *)lastOrderedDescendantInOrderedItems:(NSOrderedSet *)orderedSet;
```

Comment:
Great feedback. I renamed the method and added some documentation. Let me know if it reads well to you, might need some revision.